rdf/shared/src/main/scala/org/w3/banana/RDF.scala

```diff
@@ -4,6 +4,7 @@ trait RDF {
   // types related to the RDF datamodel
   type Graph
   type Triple
+  type Quad
```

Comment:
I added the Quad definition to show how it would be defined in the concrete implementations. I know it doesn't belong to the patch, as it adds no value and would confuse others.

Originally I was thinking about adding Quads on this level, but I don't know exactly how to implement it, with preferable minimal changes in banana rdf.... Maybe I will put some afford in it later.

Reply:
yes, I understand very well. Quads make a lot of sense if what the parser returns is a stream of quads. If what it returns is a DataSet, then you don't necessarily need them. 

So looking at Jena I see they use a `PipedRDFIterator[Quad]()`. And Sesame actually has a method like ` parser.setRDFHandler(collector)` which is also something that processes the Quads one at a time. 

So I suggest instead of trying to extend RDFReader, you could try to create a new io trait `RDFStream[X]` that would parse a document and return an Input or Output Stream of Xs (either Triples or Quads). 

That is what both Jena and Sesame do.